src/os/bluestore/BlueStore.h

```diff
@@ -1649,6 +1671,31 @@ class BlueStore : public ObjectStore,
       modified_objects.insert(o);
     }
 
+    void note_new_object(OnodeRef &o) {
+      ceph_assert(onode_to_offset_map.count(o) == 0);
```

Comment:
You might want to use insert or emplace call which return <iterator/success flag> pair instead. And assert on this flag if no success. Hence doing single lookup instead of two: for count() and for the below assignment.

Reply:
Done.